protonj2/src/main/java/org/apache/qpid/protonj2/codec/encoders/AbstractDescribedMapTypeEncoder.java

```diff
@@ -156,8 +156,8 @@ public void writeArray(ProtonBuffer buffer, EncoderState state, Object[] values)
         // Move back and write the size
         final int writeSize = buffer.getWriteIndex() - startIndex - Integer.BYTES;
 
-        if (writeSize > Integer.MAX_VALUE) {
-            throw new IllegalArgumentException("Cannot encode given array, encoded size to large: " + writeSize);
+        if (writeSize > Integer.MAX_VALUE) {  // same, always false
```

Comment:
condition is always false

Reply:
Not a spelling error

Reply:
I've rebased out the comment; do you think the "condition is always false" is worth addressing somehow? I would think so, given the if check directly before it...

Reply:
There is no breakage caused by it and a future change to buffer API might allow for long values to allow for very large composite buffers at which point this would be valid so not sure I would address it here.

Reply:
Makes sense, thanks, I was thinking that you maybe meant to use long there right away... Ok, I'm leaving this as it is.